### PR TITLE
Add no-op context menu item for deploy

### PR DESCRIPTION
**Pull request by akerekes**
Related to #283 , implements item 2)

### CODE REVIEW

**Comment by akerekes** [comment]
@nbashirbello PTAL

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/plugin.xml [hunk 1]
```diff
@@ -1,5 +1,32 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <?eclipse version="3.4"?>
 <plugin>
+   <extension
+         point="org.eclipse.ui.commands">
+      <command
+            description="Deploy the project to App Engine Standard runtime"
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/plugin.xml]: put this user visible string in a .properties file

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/plugin.xml [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+            id="com.google.cloud.tools.eclipse.appengine.deploy.standard"
+            name="Deploy to App Engine Standard">
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/plugin.xml]: put this user visible string in a .properties file

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/plugin.xml [hunk 3]
```diff
@@ -1,5 +1,32 @@
...
+      </command>
+   </extension>
+   <extension
+         point="org.eclipse.ui.menus">
+      <menuContribution
+            allPopups="false"
+            locationURI="popup:org.eclipse.ui.popup.any?after=additions">
+         <command
+               commandId="com.google.cloud.tools.eclipse.appengine.deploy.standard"
+               label="Deploy to App Engine Standard"
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/plugin.xml]: put this user visible string in a .properties file

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.ui.handlers.HandlerUtil;
+
+public class AppEngineStandardDeployCommandHandler extends AbstractHandler {
+
+  @Override
+  public Object execute(ExecutionEvent event) throws ExecutionException {
+    new MessageDialog(HandlerUtil.getActiveShell(event), "Deploying to App Engine Standard", null,
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java]: You can use the Eclipse Externalize Strings Wizard to make a resource bundle for this.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.ui.handlers.HandlerUtil;
+
+public class AppEngineStandardDeployCommandHandler extends AbstractHandler {
+
+  @Override
+  public Object execute(ExecutionEvent event) throws ExecutionException {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java]: Is it plausible to test this class at all?

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java (hunk 4 above)
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java]: Done, although this dialog will almost certainly not make it to the final code, but the code to use externalized strings are going to be useful later.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java (hunk 5 above)
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java]: Right now, I don't think it'd make much sense as this is just a dummy implementation, once something meaningful is there, we can look into it.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/plugin.properties [hunk 6]
```diff
@@ -1,4 +1,8 @@
 cloudsdkdeploy=App Engine Project Deploy
 
+deployStandardCommandDescription=Deploy the project to App Engine Standard runtime
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/plugin.properties]: delete "runtime". That's a local concept.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java (hunk 5 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java]: Per Javadoc the execute method is deprecated. Looks like you're supposed to call fireHandlerChanged instead?

http://help.eclipse.org/kepler/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fui%2Fcommands%2FAbstractHandler.html

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    new MessageDialog(HandlerUtil.getActiveShell(event), Messages.getString("deploy.standard.dialog.title"), null, //$NON-NLS-1$
+      Messages.getString("deploy.standard.dialog.message"), //$NON-NLS-1$
+      MessageDialog.INFORMATION, new String[]{Messages.getString("button.ok.label")}, 0).open(); //$NON-NLS-1$
+    return null;
+  }
+
+  @Override
+  public boolean isEnabled() {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java]: I'm tempted to test at least this much now: skeletal code and skeletal test will give us something to build on later.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java (hunk 5 above)
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java]: I'm using here the new AbstractHandler from the org.eclipse.core.commands bundle as instructed by the deprecation message.
Looking at how classes in the platform implement handlers, `execute()` is still the method to use. `fireHandlerChanged` is used when the handler changes, e.g. `isEnabled()` changes, or some other (not yet existing) property of the handler.

**Comment by elharo** [comment]
LGTM

**Pull request merged**